environs/networking.go

```diff
@@ -55,7 +55,7 @@ type Networking interface {
 	// Spaces returns a slice of network.SpaceInfo with info, including
 	// details of all associated subnets, about all spaces known to the
 	// provider that have subnets available.
-	Spaces(ctx context.ProviderCallContext) ([]network.SpaceInfo, error)
+	Spaces(ctx context.ProviderCallContext) (network.SpaceInfos, error)
```

Comment:
So I know we've found that it is much better to have local interface definitions, rather than using a shared one.
Is it useful to have a shared type here? I suppose it doesn't provide more coupling, since we already depend on something from network.